website/client-old/js/controllers/inventoryCtrl.js

```diff
@@ -253,20 +253,40 @@ habitrpg.controller("InventoryCtrl",
             size:'sm'
           });
         }
-
-        // Checks if mountmaster has been reached for the first time
-        if(!user.achievements.mountMaster
-            && $scope.mountCount >= 90) {
-          User.user.achievements.mountMaster = true;
-          Achievement.displayAchievement('mountMaster');
-        }
+        handleMountAchievements();
 
       // Selecting Pet
       } else {
         User.equip({params:{type: 'pet', key: pet.key}});
       }
     }
+    function handleMountAchievements() {
+        //Mount collector
+        var retrospectiveCount = (user.achievements.mountMaster || 0) * 90 + $rootScope.countExists(user.items.mounts);
+        user.achievements.mountCollector = Math.max(retrospectiveCount, user.achievements.mountCollector || 0);
```

Comment:
It might be that you're only saving to the client-local `user` instead of the server `User`. Try changing out `user.achievements.mountCollector` for `User.user.achievements.mountCollector` when updating the value. (When reading the value, as in the latter part of this line, `user.` is fine.)